src/test/UniswapV2LockstakeCallee.t.sol

```diff
@@ -0,0 +1,418 @@
+        if (withDelegate) {
+            engine.selectVoteDelegate(urn, voteDelegate);
+        }
+        if (withStaking) {
+            engine.selectFarm(urn, address(farm), 0);
+        }
+        mkr.approve(address(engine), 100_000 * 10**18);
+        engine.lock(urn, 100_000 * 10**18, 5);
+        engine.draw(urn, address(this), 2_000 * 10**18);
+        assertEq(_ink(ilk, urn), 100_000 * 10**18);
+        assertEq(_art(ilk, urn), 2_000 * 10**18);
+
+        if (withDelegate) {
+            assertEq(engine.urnVoteDelegates(urn), voteDelegate);
+            assertEq(mkr.balanceOf(voteDelegate), 100_000 * 10**18);
+            assertEq(mkr.balanceOf(address(engine)), 0);
+        } else {
+            assertEq(engine.urnVoteDelegates(urn), address(0));
+            assertEq(mkr.balanceOf(address(engine)), 100_000 * 10**18);
+        }
+        if (withStaking) {
+            assertEq(lsmkr.balanceOf(address(urn)), 0);
+            assertEq(lsmkr.balanceOf(address(farm)), 100_000 * 10**18);
+            assertEq(farm.balanceOf(address(urn)), 100_000 * 10**18);
+        } else {
+            assertEq(lsmkr.balanceOf(address(urn)), 100_000 * 10**18);
+        }
+    }
+
+    // Match https://github.com/makerdao/lockstake/blob/735e1e85ca706534a77d8e1582df0d3248cbd2b6/test/LockstakeEngine.t.sol#L993-L1005
+    function _forceLiquidation(address urn) internal returns (uint256 id) {
+        vm.store(address(pip), bytes32(uint256(1)), bytes32(uint256(0.05 * 10**18))); // Force liquidation
+        dss.spotter.poke(ilk);
+        assertEq(clip.kicks(), 0);
+        assertEq(engine.urnAuctions(urn), 0);
+        (,, uint256 hole,) = dss.dog.ilks(ilk);
+        uint256 kicked = hole < 2_000 * 10**45 ? 100_000 * 10**18 * hole / (2_000 * 10**45) : 100_000 * 10**18;
+        vm.expectEmit(true, true, true, true);
+        emit OnKick(urn, kicked);
+        id = dss.dog.bark(ilk, address(urn), address(this));
+        assertEq(clip.kicks(), 1);
+        assertEq(engine.urnAuctions(urn), 1);
+    }
+
+    // Based on the `_testOnTake` https://github.com/makerdao/lockstake/blob/735e1e85ca706534a77d8e1582df0d3248cbd2b6/test/LockstakeEngine.t.sol#L1104-L1202
+    function _testCalleeTake(bool withDelegate, bool withStaking, address[] memory path, uint256 exchangeRate) internal {
+        // Setup urn and force its liquidation
+        address urn = _urnSetUp(withDelegate, withStaking);
+        uint256 id = _forceLiquidation(urn);
+
+        // Determine destination token
+        TokenLike dst = TokenLike(path[path.length - 1]);
+
+        // Setup buyer
+        address buyer1 = address(111);
+        vm.prank(buyer1); dss.vat.hope(address(clip));
+        assertEq(mkr.balanceOf(buyer1), 0, 'unexpected-initial-buyer1-mkr-balance');
+        assertEq(dst.balanceOf(buyer1), 0, 'unexpected-initial-buyer1-dst-balance');
+
+        // Partial profit
+        uint256 expectedProfit = (20_000 * 10**18) * exchangeRate - 20_000 * pip.read() * clip.buf() / RAY;
```

Comment:
nit,
For me, it was a bit confusing where this `20_000` came from without reading further.
I think it'd be nice if we could put this value in a var (eg. `amountToBy1 = 20_000`)